components/bitcoind/src/utils/bitcoind.rs

```diff
@@ -43,6 +47,57 @@ pub fn bitcoind_get_chain_tip(config: &BitcoindConfig, ctx: &Context) -> BlockId
     }
 }
 
+/// Retrieves the block_height for a given blockhash.
+pub fn bitcoind_get_block_height(
+    bitcoin_rpc: &Client,
+    ctx: &Context,
+    blockhash: &BlockHash,
+) -> u32 {
+    loop {
```

Comment:
I don't think these functions should attempt to retry the RPC calls, we should propagate the error instead so we can adequately handle it and retry the entire block indexing or whatever we need to do, i.e. your return types should be `Result<u32, String>`